pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -424,7 +424,7 @@ let
       MODULE_SIG            = no; # r13y, generates a random key during build and bakes it in
       # Depends on MODULE_SIG and only really helps when you sign your modules
       # and enforce signatures which we don't do by default.
-      SECURITY_LOCKDOWN_LSM = no;
+      SECURITY_LOCKDOWN_LSM = option no;
```

Comment:
I'm not familiar with how common-config works - how would this differ from, say, `optionalAttrs (versionAtLeast version "5.4") { SECURITY_LOCKDOWN_LSM = no; }` ?

Reply:
Not sure how it works precisely but optional is a function that updates the `no` attrset with `{ optional = true; }` which makes it the setting optional, thus allowing the config file to build despite the kernel not understanding the setting.

I'm pretty sure there's a difference but optionalAttrs would require some not-so-nice syntax that I'd rather avoid and this works just fine. Many other config settings use this too.

Reply:
Ok, thanks :)